topdown/crypto.go

```diff
@@ -205,6 +209,55 @@ func builtinCryptoX509ParseRSAPrivateKey(_ BuiltinContext, operands []*ast.Term,
 	return iter(ast.NewTerm(value))
 }
 
+func builtinCryptoParsePrivateKey(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
+
+	a := operands[0].Value
+	input, err := builtins.StringOperand(a, 1)
+	if err != nil {
+		return err
+	}
+
+	// get the raw private key
+	rawKey, err := getRSAPrivateKeyFromString(string(input))
```

Comment:
suggestion: I think that this function should be renamed to drop the RSA part as it's now doing more than just RSA.

Reply:
see 46e405038b8e65af976a84bf60c2c65450e42578